src/meta/src/manager/cluster.rs

```diff
@@ -226,28 +225,47 @@ where
 
     pub async fn update_schedulability(
         &self,
-        host_address: HostAddress,
+        worker_ids: &[u32],
         is_unschedulable: bool,
-    ) -> MetaResult<WorkerType> {
+    ) -> MetaResult<Vec<WorkerType>> {
         let mut core = self.core.write().await;
-        let worker = core
+        let workers = core
             .workers
-            .get_mut(&WorkerKey(host_address.clone()))
-            .ok_or_else(|| anyhow!("Worker node does not exist!"))?;
-        let worker_type = worker.worker_type();
-
-        if let Some(property) = &mut worker.worker_node.property {
-            if property.is_unschedulable == is_unschedulable {
-                return Ok(worker_type);
+            .values_mut()
+            .filter(|w| worker_ids.contains(&w.worker_id()))
+            .collect_vec();
+        if workers.is_empty() {
+            return Err(MetaError::invalid_parameter("Worker nodes do not exist!"));
```

Comment:
Should we return error if any provided worker does not exist?

Reply:
Sure. We can do that.